### PR TITLE
Document various natives and improve documentation on existing natives

### PR DESCRIPTION
**Pull request by DurtyFree**
Research & informations I have collected, credits to various people that helped or contributed on some channels

### CODE REVIEW

File: natives.json [hunk 1]
```diff
@@ -4269,7 +4269,7 @@
 		"0x0653B735BFBDFE87": {
 			"name": "_0x0653B735BFBDFE87",
 			"jhash": "",
-			"comment": "",
+			"comment": "Sets CPed+C46\nSET_PED_??",
```
Comment (Disquse) [inline natives.json:4272]: Enables/disables ped's "loud" footstep sound.

File: natives.json [hunk 2]
```diff
@@ -20354,7 +20354,7 @@
 		"0x7B226C785A52A0A9": {
 			"name": "IS_SCREENBLUR_FADE_RUNNING",
 			"jhash": "0x926B8734",
-			"comment": "",
+			"comment": "Returns is transition to blur/from blur is running.",
```
Comment (LeonMrBonnie) [inline natives.json:20357]: This sentence has wrong grammar.

File: natives.json [hunk 3]
```diff
@@ -4286,7 +4286,7 @@
 		"0x29DA3CA8D8B2692D": {
 			"name": "_0x29DA3CA8D8B2692D",
 			"jhash": "",
-			"comment": "",
+			"comment": "Sets ped+c47 (def 1)\nSET_PED_??",
```
Comment (Disquse) [inline natives.json:4289]: Enables/disables ped's "quiet" footstep sound.

File: natives.json [hunk 4]
```diff
@@ -35258,9 +35258,9 @@
 		"0x44A0BDC559B35F6E": {
 			"name": "_0x44A0BDC559B35F6E",
 			"jhash": "0x3589452B",
-			"comment": "",
+			"comment": "Show a GTA Screen who tell than before you redo a mission, the game will make a restoration point where you will cameback when the mission is over.\nReturns 1 if the message isn't currently on screen",
```
Comment (LeonMrBonnie) [inline natives.json:35261]: This sentence has wrong grammar.

File: natives.json [hunk 5]
```diff
@@ -66560,16 +66560,16 @@
 			"build": "323"
 		},
 		"0x76BBA2CEE66D47E9": {
-			"name": "_0x76BBA2CEE66D47E9",
+			"name": "_GET_PED_EYE_COLOR",
 			"jhash": "",
-			"comment": "",
+			"comment": "A getter for _SET_PED_EYE_COLOR]. Returns -1 if fails to get.",
```
Comment (LeonMrBonnie) [inline natives.json:66565]: There's a random `]` in there.

File: natives.json [hunk 6]
```diff
@@ -105367,17 +105367,17 @@
 			"build": "1290"
 		},
 		"0x94A68DA412C4007D": {
-			"name": "_0x94A68DA412C4007D",
+			"name": "_SET_CARGOBOB_HOOK_CAN_ATTACH",
 			"jhash": "",
-			"comment": "",
+			"comment": "Stops the cargobob from beeing able to attach any vehicle",
```
Comment (LeonMrBonnie) [inline natives.json:105372]: `beeing` should be `being`

File: natives.json [hunk 7]
```diff
@@ -105648,49 +105648,49 @@
...
 			],
 			"return_type": "void",
 			"build": "1290"
 		},
 		"0xC60060EB0D8AC7B1": {
-			"name": "_0xC60060EB0D8AC7B1",
+			"name": "_SET_VEHICLE_TURRET_UNK",
 			"jhash": "",
-			"comment": "",
+			"comment": "set flag on turret",
```
Comment (LeonMrBonnie) [inline natives.json:105681]: Missing capitalization

File: natives.json [hunk 8]
```diff
@@ -105921,7 +105921,7 @@
 		"0x211E95CE9903940C": {
 			"name": "_SET_DISABLE_VEHICLE_UNK_2",
 			"jhash": "",
-			"comment": "",
+			"comment": "sets some global bool",
```
Comment (LeonMrBonnie) [inline natives.json:105924]: Missing capitalization

File: natives.json [hunk 9]
```diff
@@ -89200,9 +89200,9 @@
 			"build": "2189"
 		},
 		"0xDBBC7A2432524127": {
-			"name": "_0xDBBC7A2432524127",
+			"name": "CLEAR_VEHICLE_TASKS",
```
Comment (Disquse) [inline natives.json:89203]: Is this name confirmed? Asking because there's no jhash.

Comment (DurtyFree) [reply natives.json:89203]: No (atleast not by me), not sure from where it is, I probably just missed the underscore here

All contribution here is a mix of some stuff i figured out and found on various places

File: natives.json [hunk 10]
```diff
@@ -37375,31 +37375,31 @@
 		"0x32C7A7E8C43A1F80": {
 			"name": "ADD_POP_MULTIPLIER_SPHERE",
 			"jhash": "0xCE4A3D20",
-			"comment": "",
+			"comment": "This native is adding a zone, where you can change density settings. For example, you can add a zone on 0.0, 0.0, 0.0 with radius 900.0 and vehicleMultiplayer 0.0, and you will not see any new population vehicle spawned in a radius of 900.0 from 0.0, 0.0, 0.0. Returns the id. You can have only 15 zones at the same time. You can remove zone using REMOVE_POP_MULTIPLIER_SPHERE",
```
Comment (Disquse) [inline natives.json:37378]: vehicle**Multiplier**, not multiplayer, lol

File: natives.json [hunk 11]
```diff
@@ -37416,7 +37416,7 @@
 		"0x171BAFB3C60389F4": {
 			"name": "DOES_POP_MULTIPLIER_SPHERE_EXIST",
 			"jhash": "0xD2B7A5A0",
-			"comment": "",
+			"comment": "Removes population multiplier sphere",
```
Comment (LeonMrBonnie) [inline natives.json:37419]: The name for this does not seem to match what it does, are you sure about this?

File: natives.json [hunk 12]
```diff
@@ -85756,7 +85756,7 @@
 			"build": "323"
 		},
 		"0xAD5FDF34B81BFE79": {
-			"name": "_0xAD5FDF34B81BFE79",
+			"name": "_DISABLE_SWITCH_PAUSE_BEFORE_DESCENT",
```
Comment (gottfriedleibniz) [inline natives.json]:
```
0x8e2a065abdae6994 ALLOW_PLAYER_SWITCH_ASCENT
0xad5fdf34b81bfe79 ALLOW_PLAYER_SWITCH_DESCENT
0x74de2e8739086740 ALLOW_PLAYER_SWITCH_OUTRO
0x43d1680c6d19a8e9 ALLOW_PLAYER_SWITCH_PAN
```

File: natives.json [hunk 13]
```diff
@@ -28842,7 +28842,7 @@
 		"0x72C1056D678BB7D8": {
 			"name": "_HUD_WEAPON_WHEEL_SET_SLOT_HASH",
 			"jhash": "0x83B608A0",
-			"comment": "",
+			"comment": "Seems to automatically select the weapon on the corresponding slot in the weapon selection wheel",
```
Comment (gottfriedleibniz) [inline natives.json:28845]:
```
0x72c1056d678bb7d8 HUD_SET_WEAPON_WHEEL_TOP_SLOT
```

File: natives.json [hunk 14]
```diff
@@ -85870,13 +85870,13 @@
 			"build": "323"
 		},
 		"0xA76359FC80B2438E": {
-			"name": "_0xA76359FC80B2438E",
+			"name": "_SET_TEXTURE_DISTANCE_SCALING_THIS_FRAME",
```
Comment (gottfriedleibniz) [inline natives.json]:
```
0x0C15B0E443B2349D GET_LODSCALE
0xA76359FC80B2438E OVERRIDE_LODSCALE_THIS_FRAME
```

File: natives.json [hunk 15]
```diff
@@ -14580,7 +14580,7 @@
 			"build": "323"
 		},
 		"0xCE6294A232D03786": {
-			"name": "_GET_ENTITY_BONE_ROTATION",
+			"name": "_GET_WORLD_ROTATION_OF_ENTITY_BONE",
```
Comment (gottfriedleibniz) [inline natives.json:14583]: This PR will never be merged if you continue to synchronize the many mistakes (or in this particular case: an outdated name) of the FiveM native repo with this repository. ... 

Break it up into more manageable/maintainable pieces.

Comment (DurtyFree) [reply natives.json:14583]: Its hard to judge whether a name is outdated or not as many sources have different names and as long as its not confirmed, the name describing what the native does the most is a good name I guess.

Splitting so many changes into multiple PRs isn't something I am planning to do, but anybody else is free to do it

**Comment by alloc8or** [comment]
Due to this being an unstructured mess, it'll take a while for this to be accepted/merged. It will need A LOT of changes to meet my (quality) standards and to fix all the incorrect changes. I can probably start reviewing this in the next few days.
Also, thanks to Disquse, LeonMrBonnie and gottfriedleibniz who already pointed out some of the obvious mistakes.

File: natives.json [hunk 16]
```diff
@@ -105612,85 +105616,85 @@
...
 			],
-			"return_type": "Any",
+			"return_type": "BOOL",
 			"build": "1290"
 		},
 		"0xAA653AE61924B0A0": {
 			"name": "_0xAA653AE61924B0A0",
 			"jhash": "",
-			"comment": "",
+			"comment": "Sets veh+1f86\n_SET_VEHICLE_?",
```
Comment (alloc8or) [inline natives.json:105668]: Useless offset

File: natives.json [hunk 17]
```diff
@@ -105733,11 +105737,11 @@
 		"0x887FA38787DE8C72": {
 			"name": "_0x887FA38787DE8C72",
 			"jhash": "",
-			"comment": "",
+			"comment": "sets veh+363=1\nSET_VEHICLE_??",
```
Comment (alloc8or) [inline natives.json:105740]: Useless offset

File: natives.json [hunk 18]
```diff
@@ -105832,7 +105836,7 @@
 		"0xE8718FAF591FD224": {
 			"name": "_0xE8718FAF591FD224",
 			"jhash": "",
-			"comment": "GET_H*",
+			"comment": "Checks if float at CVehicle+0x364 is 0\n_IS_VEHICLE_??",
```
Comment (alloc8or) [inline natives.json:105839]: Undo this change entirely please

File: natives.json [hunk 19]
```diff
@@ -105945,7 +105949,7 @@
 			"build": "1604"
 		},
 		"0x71AFB258CCED3A27": {
-			"name": "_GET_DOES_VEHICLE_HAVE_TOMBSTONE",
+			"name": "_GET_IS_VEHICLE_HAS_DETACHABLE_BUMPER",
```
Comment (alloc8or) [inline natives.json:105952]: Undo

File: natives.json [hunk 20]
```diff
@@ -105958,9 +105962,9 @@
 			"build": "1604"
 		},
 		"0xAE71FB656C600587": {
-			"name": "_HIDE_VEHICLE_TOMBSTONE",
+			"name": "_SET_VEHICLE_DETACHABLE_BUMPER_DISABLED",
```
Comment (alloc8or) [inline natives.json:105965]: Undo

File: natives.json [hunk 21]
```diff
@@ -106848,7 +106852,7 @@
 					"name": "equipNow"
 				}
 			],
-			"return_type": "void",
+			"return_type": "int",
```
Comment (alloc8or) [inline natives.json:106855]: This native does not return anything.

File: natives.json [hunk 22]
```diff
@@ -106997,7 +107001,7 @@
 		"0x0E98F88A24C5F4B8": {
 			"name": "CLEAR_PED_LAST_WEAPON_DAMAGE",
 			"jhash": "0x52C68832",
-			"comment": "",
+			"comment": "Does NOT seem to work with HasPedBeenDamagedByWeapon. Use CLEAR_ENTITY_LAST_WEAPON_DAMAGE and HAS_ENTITY_BEEN_DAMAGED_BY_WEAPON instead.",
```
Comment (alloc8or) [inline natives.json:107004]: FiveM format

**Comment by DurtyFree** [comment]
@alloc8or Please tag me if you are done reviewing it and let me know what to do with the "FiveM format" conversations. (Remove comment entirely?)

**Comment by Disquse** [comment]
> 
> 
> @alloc8or Please tag me if you are done reviewing it and let me know what to do with the "FiveM format" conversations. (Remove comment entirely?)

Pretty sure he wants "FiveM format" to be changed into upper case format. I.e. `GetEntityCoords` -> `GET_ENTITY_COORDS`.

**Comment by alloc8or** [comment]
^ what Disquse said.
I didn't have much time and tried to be as quick as I can, sorry for the confusion.

File: natives.json [hunk 23]
```diff
@@ -87367,17 +87371,17 @@
 			"build": "323"
 		},
 		"0x88E32DB8C1A4AA4B": {
-			"name": "_0x88E32DB8C1A4AA4B",
+			"name": "SET_PED_PATH_CLIMB_COST_MODIFIER",
```
[the remark this answers is not in the document]
Comment (gottfriedleibniz) [reply natives.json:87374]: If I remember correctly, navcapabilities.meta > fClimbCostModifier.  Should include a note that `p2` is bounded between [0.0, 10.0]

File: natives.json [hunk 24]
```diff
@@ -67580,7 +67584,7 @@
...
+			"comment": "p4/p5: Unusued in TU27\nRagdoll Types:\n**0**: CTaskNMRelax\n**1**: CTaskNMScriptControl: Hardcoded not to work in networked environments.\n**Else**: CTaskNMBalance\ntime1- Time(ms) Ped is in ragdoll mode; only applies to ragdoll types 0 and not 1.\n\ntime2- Unknown time, in milliseconds\n\nragdollType-\n0 : Normal ragdoll\n1 : Falls with stiff legs/body\n2 : Narrow leg stumble(may not fall)\n3 : Wide leg stumble(may not fall)\n\np4, p5, p6- No idea. In R*'s scripts they are usually either \"true, true, false\" or \"false, false, false\".\n\n\n\n\nEDIT 3/11/16: unclear what 'mircoseconds' mean-- a microsecond is 1000x a ms, so time2 must be 1000x time1?  more testing needed.  -sob\n\nEdit Mar 21, 2017: removed part about time2 being the microseconds version of time1. this just isn't correct. time2 is in milliseconds, and time1 and time2 don't seem to be connected in any way.",
```
[the remark this answers is not in the document]
Comment (gottfriedleibniz) [reply natives.json:67587]: I don't remember when this native was obfuscated, but you can also cross-reference v323 for the ragdoll types. Not entirely sure if the parameter usage has changed (and not particularly interested in checking either...)

File: natives.json [hunk 25]
```diff
@@ -7536,7 +7536,7 @@
 		"0x9DFE13ECDC1EC196": {
 			"name": "_0x9DFE13ECDC1EC196",
 			"jhash": "0x92302899",
-			"comment": "SET_FOLLOW_*",
+			"comment": "SET_FOLLOW_*\nSET_TIME_IDLE_DROP?",
```
[the remark this answers is not in the document]
Comment (gottfriedleibniz) [reply natives.json:7539]: This comment is only partially related (share common functions, but:

``
0xD1F8363DFAD03848 SET_GAMEPLAY_HINT_BASE_ORBIT_PITCH_OFFSET
0xF8BDBF3D573049A1 SET_GAMEPLAY_HINT_FOLLOW_DISTANCE_SCALAR
// likely fields of camHintHelperMetadata: 0xFC7464A0, 0x3554AA0E, 0x2F0CE859
``

File: natives.json [hunk 26]
```diff
@@ -31051,11 +31051,11 @@
 			"build": "323"
 		},
 		"0x2E22FEFA0100275E": {
-			"name": "_0x2E22FEFA0100275E",
+			"name": "_GET_MENU_STATE_CHANGED_THIS_FRAME",
```
[the remark this answers is not in the document]
Comment (gottfriedleibniz) [reply natives.json:31054]: I'd considering undoing this change until the name of that group of natives has been discovered, see: 0x7e17be53e1aaabaf and 0x36c1451a88a09630 (which the parameters look like they are pointers that store the last input for each input group during menu selection jumping?), and everything in between.

[the remark this answers is not in the document]
Comment (alloc8or) [reply natives.json:31054]: Good point, I agree.

File: natives.json [hunk 27]
```diff
@@ -36698,8 +36698,8 @@
 			"comment": "Formerly known as _LOWER_MAP_PROP_DENSITY and wrongly due to idiots as _ENABLE_MP_DLC_MAPS.\nSets the maximum prop density and changes a loading screen flag from 'loading story mode' to 'loading GTA Online'. Does not touch DLC map data at all.\n\nIn fact, I doubt this changes the flag whatsoever, that's the OTHER native idiots use together with this that does so, this one only causes a loading screen to show as it reloads map data.",
 			"params": [
 				{
-					"type": "int",
-					"name": "p0"
+					"type": "BOOL",
+					"name": "toggle"
```
[the remark this answers is not in the document]
Comment (alloc8or) [reply natives.json:36702]: You forgot to change the param name. Since this is an '``int`` it certainly is not a "toggle".

File: natives.json [hunk 28]
```diff
@@ -38291,7 +38291,7 @@
 		"0xE574A662ACAEFBB1": {
 			"name": "_RESTART_GAME",
 			"jhash": "",
-			"comment": "",
+			"comment": "Shows `Disconnecting from GTA Online` HUD and then quits the game. Better name `_QUIT_GAME`?",
```
[the remark this answers is not in the document]
Comment (alloc8or) [reply natives.json:38294]: It should be noted that it does exactly what the name implies in SP.
Also, remove the "Better name \`_QUIT_GAME\`?" part.

File: natives.json [hunk 29]
```diff
@@ -50533,7 +50533,7 @@
 		"0x9ECA15ADFE141431": {
 			"name": "_SHUTDOWN_AND_LOAD_MOST_RECENT_SAVE",
 			"jhash": "",
-			"comment": "",
+			"comment": "Disconnects you from the session, and starts loading single player, however you still remain connected to the server (only if you're the host, if you're not then you also (most likely) get disconnected from the server) and other players will not be able to join until you exit the game.\n\nIf you're already in SP then it'll re-load singleplayer.\n\nYou might need to DoScreenFadeIn and ShutdownLoadingScreen otherwise you probably won't end up loading into SP at all.\n\nSomewhat related note: opening the pause menu after loading into this 'singleplayer' mode crashes the game.",
```
[the remark this answers is not in the document]
Comment (alloc8or) [reply natives.json:50536]: Again, could be slightly confusing the way it's worded at the moment. I'd suggest to move the part about SP to the beginning and then mention the GTAO/FiveM specific behavior.

File: natives.json (hunk 23 above)
[the remark this answers is not in the document]
Comment (alloc8or) [reply natives.json:87374]: ``Could be the move speed on the path. Needs testing.\n\n`` can be removed then, I guess.

**Comment by DurtyFree** [comment]
@alloc8or Applied all requested changes, thank you for everyones input.

**Comment by gottfriedleibniz** [comment]
The comment sections to `0xF7B38B8305F1FE8B`, `0x0C5A80A9E096D529`, `0x3CA6050692BC61B0`, etc are technically invalid JSON (double quotes need to be escaped). I'd probably run these changes through a JSON tool beforehand.

Also, other related hashes: 
```
0xD1F8363DFAD03848 SET_GAMEPLAY_HINT_BASE_ORBIT_PITCH_OFFSET
0xF8BDBF3D573049A1 SET_GAMEPLAY_HINT_FOLLOW_DISTANCE_SCALAR
// likely fields of camHintHelperMetadata: 0xFC7464A0, 0x3554AA0E, 0x2F0CE859
```
